cmd/image/qcow2ova/prep/loop.go

```diff
@@ -0,0 +1,86 @@
+package prep
+
+import (
+	"fmt"
+	"github.com/ppc64le-cloud/pvsadm/pkg/utils"
+	"strings"
+)
+
+const losetupCMD = "losetup"
+
+func findFreeLoop(file string) (string, error) {
+	exitcode, out, err := utils.RunCMD(losetupCMD, "-f")
+	if exitcode != 0 {
+		return "", fmt.Errorf("failed to find an unused loop device, exitcode: %d, stdout: %s, err: %s", exitcode, out, err)
+	}
+
+	return strings.TrimSpace(out), nil
+}
+
+// setupLoop allocates the free loop device for the backing store
+func setupLoop(file string) (string, error) {
+	exitcode, out, err := utils.RunCMD(losetupCMD, "-f", "--show", file)
+	if exitcode != 0 {
+		return "", fmt.Errorf("failed to setup a loop device for file: %s, exitcode: %d, stdout: %s, err: %s", file, exitcode, out, err)
+	}
+
+	return strings.TrimSpace(out), nil
+}
+
+func partprobe(device string) error {
+	exitcode, out, err := utils.RunCMD("partprobe", device)
+	if exitcode != 0 {
+		return fmt.Errorf("failed to partprobe: %s, exitcode: %d, stdout: %s, err: %s", device, exitcode, out, err)
+	}
+	return nil
+}
+
+// growpart resizes the partition
+func growpart(device, partition string) error {
+	exitcode, out, err := utils.RunCMD("growpart", device, partition)
+	if exitcode != 0 {
+		return fmt.Errorf("failed to growpart for the device: %s and partition: %s, exitcode: %d, stdout: %s, err: %s", device, partition, exitcode, out, err)
+	}
+	return nil
+}
+
+// getFSType returns the filesystem type for the given device and the partition
+func getFSType(device string) (string, error) {
+	exitcode, out, err := utils.RunCMD("blkid", device, "-o", "value", "-s", "TYPE")
+	if exitcode != 0 {
+		return "", fmt.Errorf("failed to get the filesystem type for the device: %s, exitcode: %d, stdout: %s, err: %s", device, exitcode, out, err)
+	}
+	return strings.TrimSpace(out), nil
+}
+
+func xfsGrow(device string) error {
+	exitcode, out, err := utils.RunCMD("xfs_growfs", "-d", device)
+	if exitcode != 0 {
+		return fmt.Errorf("failed to xfs_growfs device: %s, exitcode: %d, stdout: %s, err: %s", device, exitcode, out, err)
+	}
+	return nil
+}
+
+func resize2fs(device string) error {
+	exitcode, out, err := utils.RunCMD("resize2fs", device)
+	if exitcode != 0 {
+		return fmt.Errorf("failed to resize2fs the device: %s, exitcode: %d, stdout: %s, err: %s", device, exitcode, out, err)
+	}
+	return nil
+}
+
+func mount(opts, src, target string) error {
+	exitcode, out, err := utils.RunCMD("mount", "-o", opts, src, target)
+	if exitcode != 0 {
+		return fmt.Errorf("failed to bind mount, source: %s and target: %s, exitcode: %d, stdout: %s, err: %s", src, target, exitcode, out, err)
+	}
+	return nil
+}
+
+func Umount(dir string) error {
```

Comment:
should this be in smallcaps to align with how you have handled the other commands - mount, resize2fs etc.. ?

Reply:
This was exposed outside because I was using this outside of this package scope in the defer call here https://github.com/mkumatag/pvsadm/blob/qcow2ova/cmd/image/qcow2ova/qcow2ova.go#L125